src/Symfony/Component/Yaml/Parser.php

```diff
@@ -1137,106 +1141,148 @@ private function getLineTag(string $value, int $flags, bool $nextLineCheck = tru
-            throw new \InvalidArgumentException(sprintf('"%s" is not a quoted string.', $yaml));
-        }
-
-        $lines = [$yaml];
+        $quotation = $this->currentLine[$cursor];
+        $value = $quotation;
+        ++$cursor;
 
-        while ($this->moveToNextLine()) {
-            $lines[] = $this->currentLine;
+        $previousLineWasNewline = true;
+        $previousLineWasTerminatedWithBackslash = false;
 
-            if (!$this->isCurrentLineEmpty() && $yaml[0] === $this->currentLine[-1]) {
-                break;
-            }
-        }
-
-        $value = '';
-
-        for ($i = 0, $linesCount = \count($lines), $previousLineWasNewline = false, $previousLineWasTerminatedWithBackslash = false; $i < $linesCount; ++$i) {
-            if ('' === trim($lines[$i])) {
+        do {
+            if ($this->isCurrentLineBlank()) {
                 $value .= "\n";
             } elseif (!$previousLineWasNewline && !$previousLineWasTerminatedWithBackslash) {
                 $value .= ' ';
             }
 
-            if ('' !== trim($lines[$i]) && '\\' === substr($lines[$i], -1)) {
-                $value .= ltrim(substr($lines[$i], 0, -1));
-            } elseif ('' !== trim($lines[$i])) {
-                $value .= trim($lines[$i]);
+            for (; \strlen($this->currentLine) > $cursor; ++$cursor) {
+                switch ($this->currentLine[$cursor]) {
+                    case '\\':
+                        if (isset($this->currentLine[++$cursor])) {
+                            $value .= '\\'.$this->currentLine[$cursor];
+                        }
+
+                        break;
+                    case $quotation:
+                        ++$cursor;
+
+                        if ("'" === $quotation && isset($this->currentLine[$cursor]) && "'" === $this->currentLine[$cursor]) {
+                            $value .= "''";
+                            break;
+                        }
+
+                        return $value.$quotation;
+                    default:
+                        $value .= $this->currentLine[$cursor];
+                }
             }
 
-            if ('' === trim($lines[$i])) {
+            if (isset($this->currentLine[$cursor]) && $this->currentLine[$cursor] === $quotation) {
+                ++$cursor;
+
+                return $value.$quotation;
+            }
+
+            if ($this->isCurrentLineBlank()) {
                 $previousLineWasNewline = true;
                 $previousLineWasTerminatedWithBackslash = false;
-            } elseif ('\\' === substr($lines[$i], -1)) {
+            } elseif ('\\' === $this->currentLine[-1]) {
                 $previousLineWasNewline = false;
                 $previousLineWasTerminatedWithBackslash = true;
             } else {
                 $previousLineWasNewline = false;
                 $previousLineWasTerminatedWithBackslash = false;
             }
-        }
+
+            $cursor = 0;
```

Comment:
@xabbuh Regarding my segmentation fault. In a case with no ending quote like `[["\\"]]` the code runs into an infinite loop because of this line. Because this yaml had just one line the situation was just worse.

There should be no parseable result if we end the while loop without returning before, right?
Either the cursor needs to keep its value at the end or better directly throw an exception?!

Reply:
Wow, thanks for testing the patch so carefully @simonberger. That's really appreciated. I have fixed this issue as well and now throw a `ParseException` as soon as possible.